### PR TITLE
aspects: support deleting entries

### PR DESCRIPTION
**Pull request by MiguelPires**
Adds support for deleting entries using the `Set` method with a nil value.

### CODE REVIEW

File: aspects/aspects.go [hunk 1]
```diff
@@ -507,7 +519,14 @@ func get(subKeys []string, index int, node map[string]json.RawMessage, result in
 // in which case, a nested JSON object is created for each sub-key found after a dot.
 func (s JSONDataBag) Set(path string, value interface{}) error {
 	subKeys := strings.Split(path, ".")
-	_, err := set(subKeys, 0, s, value)
+
+	var err error
+	if value == nil {
+		_, err = unset(subKeys, 0, s)
```
Comment (Meulengracht) [inline aspects/aspects.go:525]: Is there a reason for combining the behaviour of the two? Also I feel like this should be documented in the function docs.

Comment (MiguelPires) [reply aspects/aspects.go:525]: Yes, it's according to the specification which defines the same methods as the current configuration (`snap get/set`). I'll add a note about nil values to the function doc

**Review by pedronis — APPROVED** [review]
thank you

**Review by mvo5 — APPROVED** [review]
Thanks, this is nice and very cleanly implemented

**Pull request merged**